controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -323,3 +358,144 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error
+	}
+	valueDynatraceApiToken := string(valueBytesDynatraceApiToken)
+
+	valueBytesDynatraceTenant, ok := secret.Data[dynatraceTenantKey]
+	if !ok {
+		return "", "", fmt.Errorf("secret did not contain key %s", dynatraceTenantKey)
+	}
+	if len(valueBytesDynatraceTenant) == 0 {
+		return "", "", fmt.Errorf("%s is empty", dynatraceTenantKey)
+	}
+	valueDynatraceTenant := string(valueBytesDynatraceTenant)
+
+	return valueDynatraceApiToken, valueDynatraceTenant, nil
+}
+
+func getDynatraceHTTPMonitorID(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, error) {
+	labels := hostedcontrolplane.GetLabels()
+	dynatraceHttpMonitorId, exists := labels[httpMonitorLabel]
+	if exists {
+		return dynatraceHttpMonitorId, nil
+	}
+	return "", fmt.Errorf("key '%s' not found in labels", httpMonitorLabel)
+}
+
+func (r *HostedControlPlaneReconciler) UpdateHostedControlPlaneLabels(ctx context.Context, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane, key, value string) error {
+	labels := hostedcontrolplane.GetLabels()
+	labels[key] = value
+	hostedcontrolplane.SetLabels(labels)
+
+	err := r.Client.Update(ctx, hostedcontrolplane)
+	if err != nil {
+		return fmt.Errorf("error updating hostedcontrolplane monitor: %v", err)
+	}
+	return nil
+}
+
+func GetAPIServerHostname(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, error) {
+	for _, service := range hostedcontrolplane.Spec.Services {
+		if service.Service == "APIServer" {
+			return service.ServicePublishingStrategy.Route.Hostname, nil
+		}
+	}
+	return "", fmt.Errorf("APIServer service not found in the hostedcontrolplane")
+}
+
+func (r *HostedControlPlaneReconciler) deployDynatraceHTTPMonitorResources(DynatraceAPIClient *dynatrace.DynatraceAPIClient, ctx context.Context, log logr.Logger, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) error {
+	//if http monitor does not exist, and hcp is not marked for deletion, and hcp is ready, then create http monitor
+	//get apiserver
+	APIServerHostname, err := GetAPIServerHostname(hostedcontrolplane)
+	if err != nil {
+		return fmt.Errorf("failed to get APIServer hostname %v", err)
+	}
+	monitorName := strings.Replace(APIServerHostname, "api.", "", 1)
+	// APIServerHostname := hostedcontrolplane.Spec.Services[1].ServicePublishingStrategy.Route.Hostname
+	monitorLocation := hostedcontrolplane.Spec.Platform.AWS.EndpointAccess
+
+	//in hcp, spec.services.service["APIServer"].servicePublishingStrategy.route.hostname is api.test-rs1.dgcj.i3.devshift.org
+	// apiUrl := "https://api.hb-testing.j1b6.i3.devshift.org/livez"
+
+	apiUrl := fmt.Sprintf("https://%s/livez", APIServerHostname)
+
+	dynatraceHttpMonitorId, err := getDynatraceHTTPMonitorID(hostedcontrolplane)
+	if err != nil {
+		log.Info(fmt.Sprintf("error calling getDynatraceHTTPMonitorID %v", err))
+	}
+
+	if dynatraceHttpMonitorId != "" {
+		log.Info("HTTP monitor Found. Skipping creating a monitor")
+		return nil
+	}
```

Comment:
```suggestion
	dynatraceHttpMonitorId, ok := getDynatraceHTTPMonitorID(hostedcontrolplane)
	if !ok {
		log.Info("HTTP monitor label found. Skipping creating a monitor")
		return nil
	}
```

Reply:
Actually the if condition should be 'if ok' as we want to Skip creating a monitor if HTTP monitor label is found on hcp.
`dynatraceHttpMonitorId, ok := getDynatraceHttpMonitorId(hostedcontrolplane)
	if ok {
		log.Info(fmt.Sprintf("HTTP monitor label found. Skipping creating a monitor. Monitor id is %s", dynatraceHttpMonitorId))
		return nil
	}`

Reply:
Now we are checking if monitor also exists in dynatrace - 
```go
if ok {
		exists, err := dynatraceApiClient.ExistsHttpMonitorInDynatrace(dynatraceHttpMonitorId)
		if err != nil {
			return fmt.Errorf("failed calling ExistsHttpMonitorInDynatrace: %v", err)
		}
		if exists {
			log.Info(fmt.Sprintf("HTTP monitor label found. Skipping creating a monitor. Monitor id is %s", dynatraceHttpMonitorId))
			return nil
		}
	}
```